### PR TITLE
[@types/node] Update types for `node:console`

### PR DESCRIPTION
**Pull request by ghost**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `pnpm test node`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes:  
   - https://nodejs.org/docs/latest-v20.x/api/console.html
   - https://nodejs.org/docs/latest-v18.x/api/console.html
   - https://nodejs.org/docs/latest-v16.x/api/console.html
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

**Review by Anonymous4078 — APPROVED** [review]

**Comment by peterblazejewicz** [comment]
Does it add missing JDoc or does it add missing contract?

**Comment by codershiba** [comment]
> Does it add missing JDoc or does it add missing contract?

The PR brings the docs in sync with the Node.js docs. Also it replaced the `InspectOptions` interface being used in the `console.dir()` method with a new interface named `ConsoleDirOptions`

**Comment by peterblazejewicz** [comment]
>  Also it replaced the InspectOptions interface being used in the console.dir() method with a new interface named ConsoleDirOptions.

So please amend the title and description, thx!

**Comment by codershiba** [comment]
> > Also it replaced the InspectOptions interface being used in the console.dir() method with a new interface named ConsoleDirOptions.
> 
> So please amend the title and description, thx!

Sure

File: types/node/console.d.ts [hunk 1]
```diff
@@ -150,11 +172,11 @@ declare module "node:console" {
              */
             debug(message?: any, ...optionalParams: any[]): void;
             /**
-             * Uses `util.inspect()` on `obj` and prints the resulting string to `stdout`.
+             * Uses [`util.inspect()`](https://nodejs.org/docs/latest-v20.x/api/util.html#utilinspectobject-options) on `obj` and prints the resulting string to `stdout`.
              * This function bypasses any custom `inspect()` function defined on `obj`.
              * @since v0.1.101
              */
-            dir(obj: any, options?: InspectOptions): void;
+            dir(obj: any, options?: ConsoleDirOptions): void;
```
Comment (Semigradsky) [inline types/node/console.d.ts:179]: By the comment (Uses `util.inspect()` ...) and by the code (https://github.com/nodejs/node/blob/025ba26e517218bc744f580a7fbfa4bdebec0d0a/lib/internal/console/constructor.js#L388-L394) I see that options for `dir` the same as options for `util.inspect`. I don't think that this change is needed.

**Review by Semigradsky — CHANGES_REQUESTED** [review]
See comment

**Review by Anonymous4078 — APPROVED** [review]

**Review by gabritto — APPROVED** [review]

**Comment by codershiba** [comment]
Ready to merge

**Pull request merged**